docs/details/dispatch.md

```diff
@@ -1,5 +1,9 @@
 # Dispatch
```

Comment:
Sorry, unrelated aside: `dispatch` is a clever feature!

Perhaps this is already on your mind, but SMT generally feels like it could help with optimization here. (as opposed to doing full synthesis, you could just play with re-orderings of checks, optimizing out dead branches etc)